<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MemoryEditorWindow</class>
 <widget class="QWidget" name="MemoryEditorWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1130</width>
    <height>474</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Memory Editor</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,0">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Address:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="address"/>
       </item>
      </layout>
     </item>
     <item>
      <widget class="MemoryViewWidget" name="memoryView" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QRadioButton" name="memoryRegionRAM">
         <property name="text">
          <string>RAM</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">memoryRegionButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="memoryRegionScratchpad">
         <property name="text">
          <string>Scratchpad</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">memoryRegionButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="memoryRegionEXP1">
         <property name="text">
          <string>EXP1</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">memoryRegionButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="memoryRegionBIOS">
         <property name="text">
          <string>BIOS</string>
         </property>
         <attribute name="buttonGroup">
          <string notr="true">memoryRegionButtonGroup</string>
         </attribute>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLineEdit" name="memorySearchString"/>
       </item>
       <item>
        <widget class="QPushButton" name="memorySearch">
         <property name="text">
          <string>Search</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="dataInspector">
     <property name="title">
      <string>Data Inspector</string>
     </property>
     <layout class="QGridLayout" name="dataInspectorLayout">
      <item row="11" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>ASCII Character:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="dataInspectorUnsignedWord">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="12" column="1">
       <widget class="QLineEdit" name="dataInspectorUTF8String">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="dataInspectorUnsignedHalfword">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Signed Byte:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="dataInspectorSignedByte">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="14" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QRadioButton" name="dataInspectorDecimal">
          <property name="text">
           <string>Decimal</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">baseButtonGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="dataInspectorHexadecimal">
          <property name="text">
           <string>Hexadecimal</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">baseButtonGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="dataInspectorOctal">
          <property name="text">
           <string>Octal</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">baseButtonGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Unsigned Byte:</string>
        </property>
       </widget>
      </item>
      <item row="13" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QRadioButton" name="dataInspectorLittleEndian">
          <property name="text">
           <string>Little Endian</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">endianButtonGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="dataInspectorBigEndian">
          <property name="text">
           <string>Big Endian</string>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">endianButtonGroup</string>
          </attribute>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Unsigned Halfword:</string>
        </property>
       </widget>
      </item>
      <item row="9" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>32-Bit Float:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="dataInspectorUnsignedByte">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="dataInspectorAddress">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Unsigned Doubleword:</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QLineEdit" name="dataInspectorUnsignedDoubleWord">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="9" column="1">
       <widget class="QLineEdit" name="dataInspectorFloat32">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="dataInspectorSignedHalfword">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QLineEdit" name="dataInspectorSignedWord">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Address:</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QLineEdit" name="dataInspectorFloat64">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>UTF-8 String:</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>64-Bit Float:</string>
        </property>
       </widget>
      </item>
      <item row="11" column="1">
       <widget class="QLineEdit" name="dataInspectorASCIICharacter">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Unsigned Word:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Signed Word:</string>
        </property>
       </widget>
      </item>
      <item row="15" column="0" colspan="2">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Signed Halfword:</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>Signed Doubleword:</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QLineEdit" name="dataInspectorSignedDoubleWord">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MemoryViewWidget</class>
   <extends>QWidget</extends>
   <header>duckstation-qt/memoryviewwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources/duckstation-qt.qrc"/>
 </resources>
 <connections/>
 <buttongroups>
  <buttongroup name="baseButtonGroup"/>
  <buttongroup name="memoryRegionButtonGroup"/>
  <buttongroup name="endianButtonGroup"/>
 </buttongroups>
</ui>
